doc/lua_api.md

```diff
@@ -8103,13 +8107,20 @@ child will follow movement and rotation of that bone.
           -- Default: false
       }
       ```
+* `get_guid()`: returns a global unique identifier (a string)
+    * For players a global unique identiticator is a player name.
+    * For non-player objects, it is unique generated string.
 
 #### Lua entity only (no-op for other objects)
 
 * `remove()`: remove object
     * The object is removed after returning from Lua. However the `ObjectRef`
       itself instantly becomes unusable with all further method calls having
       no effect and returning `nil`.
+* `set_guid(guid)`:
+    * Set a global unique identifier string of entity.
+    * Is valid to be called only in `on_activate` callback.
+    * Should be used ONLY for restoring saved guid from staticdata!!!
```

Comment:
This allows GUId to be the same across multiple Lua entity loads via restoring it.

Reply:
> This allows GUId to be the same across multiple Lua entity loads

That is, in my opinion, the entire point of GUIDs, otherwise active object IDs would suffice (btw, I'd prefer GUID capitalization over GUId, which parses as.. GUI d(aemon?) for me).

The current problem roughly is that (even persistent) entities need not have GUIDs. For example a cart might want to remember which items were attached to it, to try to reattach them. [^1] Or a mob might want to remember which other mob it was currently fighting, etc.

Mods can (and do) already implement their own form(s) of GUIDs, but the problem is that this would require every mod to cooperate and agree on a GUID format (and a way to shove it into staticdata), which doesn't seem to be happening. So we want to put it in the engine, but for standardization and convenience reasons.

But then a big part of it is that persisting GUIDs **must** be done automatically, to force their existence for **all** entities without modder cooperation. `set_guid` should not exist, it just allows modders to mess up.

[^1]: Actually, I feel like this is shaping up to be a different issue related to how and in which order we load and maybe in the future persist attachments